docs(core): add astro recipe (#18277)
nrwl/nx: docs/shared/recipes/add-stack/add-astro.md

```{% command="npx nx@latest init" path="~/astro-app"%}
 >  NX   🐳 Nx initialization

 >  NX   🧑‍🔧 Please answer the following questions about the scripts found in your package.json in order to generate task runner configuration

✔ Which of the following scripts are cacheable? (Produce the same output given the same input, e.g. build, test and lint usually are, serve and start are not). You can use spacebar to select one or more scripts. · build

✔ Does the "build" script create any outputs? If not, leave blank, otherwise provide a path (e.g. dist, lib, build, coverage) · dist
✔ Enable distributed caching to make your CI faster · No

 >  NX   📦 Installing dependencies

 >  NX   🎉 Done!

   - Enabled computation caching!
   - Learn more at https://nx.dev/recipes/adopting-nx/adding-to-existing-project.
```

You can add a task as cacheable after the fact by updating the `cacheableOperations` in the `nx.json` file. Learn more about [caching task results](/recipes/adopting-nx/adding-to-existing-project#installing-nx-on-a-non-monorepo-project) or [how caching works](https://nx.dev/core-features/cache-task-results).

## Running Tasks

> Because Nx [understand package.json scripts](/reference/project-configuration#project-configuration), You can run the predefined scripts via Nx.

Reviewer: ```suggestion
Because Nx [understands package.json scripts](/reference/project-configuration#project-configuration), You can run the predefined scripts via Nx.
```

```shell
nx build
```

If you plan on using your package manager to run the tasks, then you'll want to use [`nx exec`](/packages/nx/documents/exec) to wrap the command

i.e.

```json {% fileName="package.json" %}
{
  "scripts": {
    "e2e": "nx exec -- playwright test"
  }
}
```

Now when running `npm run e2e` Nx will be able to check if there is a cache hit or not.

> If you plan to only run with Nx cli, then you can omit `nx exec` if so desired. The safest way is to always include it in the package.json script.

Reviewer: ```suggestion
If you plan to only run tasks with the Nx CLI, then you can omit `nx exec`. The safest way is to always include it in the package.json script.
```

## Using Other Plugins

With Nx plugins, you can create projects to help break out functionality of the project. For example, using the [`@nx/js:library`](https://nx.dev/packages/js/generators/library#@nx/js:library) to contain our reusable `.astro` components.

Install `@nx/js` plugin.

> Note: you should make sure any first party, `@nx/` scoped, plugins match the `nx` package version

```shell
npm i -DE @nx/js@<nx-version>
```

Then generate a project

```{% command="nx g lib --simpleName --minimal}
>  NX  Generating @nx/js:library

✔ What name would you like to use for the library? · ui
✔ Which unit test runner would you like to use? · none
✔ Which bundler would you like to use to build the library? Choose 'none' to skip build setup. · none

CREATE ui/tsconfig.json
CREATE ui/src/index.ts
CREATE ui/src/lib/ui.ts
CREATE ui/tsconfig.lib.json
CREATE ui/project.json
CREATE ui/.eslintrc.json
UPDATE tsconfig.json
```

If you plan to import `.astro` files within `.ts` files, then you can install the [`@astrojs/ts-plugin`](https://www.npmjs.com/package/@astrojs/ts-plugin).

> An easier option is to allow importing the files via the root `tsconfig.json` instead of a `index.ts` file.

Reviewer: Does `@nx/js:lib` not add a `tsconfig.base.json` file?  I would expect it to create one and add the path automatically.

Author: it adds the path to the root tsconfg.json which is from astro. but it adds the path as 
```json
"paths": {
   "@org/lib": ["lib/src/index.ts"]
}
```
which can cause issues if you import and rexport .astro files.

So you can just change it to allow importing the files directly without having to reexport and use the plugin from astro if you don't want to.

```json
"paths": {
   "@org/lib/*": ["lib/src/*"]
}
```

Reviewer: Ok, it wasn't clear that you were modifying the path that already existed.  Can you add a sentence about that?

```json {% fileName="tsconfig.json" %}
{
  "extends": "astro/tsconfigs/strict",
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "@myrepo/ui/*": ["ui/src/*"]
    }
  }
}
```

> This allows import in our `.astro` files like so

Reviewer: ```suggestion
This allows imports in our `.astro` files like so
```
